pkg/analytics/analytics.go

```diff
@@ -363,9 +363,11 @@ func getMachineID() string {
 	}
 
 	mid = generateMachineID()
-	if err := okteto.SaveMachineID(mid); err != nil {
-		log.Info("failed to save the machine id")
-		mid = "na"
+	if okteto.GetUserID() != "" {
```

Comment:
will this mean that, for users who never log in, we'll never store the machineID?

Reply:
yes, we will not store their machineID but we will retrieve it every time we need to use it.

Reply:
I don't get why we can't store the machineID though. it will a) save time (not sure how long the calculation takes) and b) ensure that it's always the same (in case something changes in the machineID calculation).

Reply:
Let's store the machineID in a different file

Reply:
That will require changes in the vscode plugin. Why do we need to store it somewhere else?

Reply:
Why does it require those changes? Creating a file named `.token.json` with no token doesn't seem right

Reply:
The file is called token.JSON but it includes more things (machineID, userID, urls, etc), so it won't be empty, it will always have the machine id.

The vscode plug-in uses the machineID  for analytics. It can read it from somewhere else, but it sounds unnecessary to have two files for this info no?